### PR TITLE
Remove usages of global databases in graphqlbackend

### PR DESCRIPTION
**Pull request by eseliger**
# 🧹

### CODE REVIEW

**Review by efritz — APPROVED** [review]

**Review by camdencheek — COMMENTED** [review]
😲 Very exciting. Not really qualified to approve, but know that I approve in my heart.

File: cmd/frontend/graphqlbackend/search_pagination.go [hunk 1]
```diff
@@ -337,13 +337,13 @@ func repoOfResult(result SearchResultResolver) string {
 //
 // If the executor returns any error, the search will be cancelled and the error
 // returned.
-func (p *repoPaginationPlan) execute(ctx context.Context, exec executor) (c *searchCursor, results []SearchResultResolver, common *streaming.Stats, err error) {
+func (p *repoPaginationPlan) execute(ctx context.Context, db dbutil.DB, exec executor) (c *searchCursor, results []SearchResultResolver, common *streaming.Stats, err error) {
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search_pagination.go:340]: nit: rather than passing in `db`, why not pass in a `database.ReposStore`?

**Review by keegancsmith — APPROVED** [review]
LGTM.

Side debate: I know we all don't like globals, because someone told us they are bad. But when _isn't_ `r.db` not the global db? IE this feels like we still have the same singleton as before, we just have the possibility of nil panics now if we forget to pass it in...

**Comment by camdencheek** [comment]
> when isn't r.db not the global db? 

Tests, ideally. I'd love to be able to pass in a mock DB without overriding things on a global level. It makes tests easier to reason about, and (in theory) faster because they can be run in parallel

**Comment by camdencheek** [comment]
Additionally, getting rid of the globals entirely means I can guarantee that, if I don't pass a DB into a method, it can't call the DB. I personally find this really useful for reasoning about chunks of code. It's less useful in `graphqlbackend` since our resolvers all hold a db instance on their struct, so any methods can quietly use it. 

This is part of the reason I've been putting so much effort into ripping references to `db` out of as much as possible. Right now, pretty much everything holds a reference to `db`, but it's not always necessary. Same way I think about context. If I don't pass a context into a function, its scope should (ideally) stay entirely in the current process.

**Review by asdine — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]
Thank you!

**Comment by efritz** [comment]
> But when isn't `r.db` not the global db?

@keegancsmith: We have three database instances.

**Pull request merged**